### PR TITLE
Network Policy for OpenMetadata

### PR DESCRIPTION
**Pull request by Anuj359**
### Describe your changes :
Worked on Kubernetes deployment #9735

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developer) document.
- [x] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: charts/openmetadata/templates/networkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{{- if .Values.networkPolicy.enabled }}
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: {{ include "OpenMetadata.fullname" . }}-networkpolicy
+  namespace: {{ .Release.Namespace | quote }}
```
Comment (akash-jain-10) [inline charts/openmetadata/templates/networkpolicy.yaml:6]: This is not required. Let's remove this! Helm install / upgrade will take care of deploying this manifest in the relevant namespaces where openmetadata is being installed!

File: charts/openmetadata/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  labels:
+    {{- include "OpenMetadata.labels" . | indent 4 }}
+  {{- with .Values.service.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
```
Comment (akash-jain-10) [inline charts/openmetadata/templates/networkpolicy.yaml:12]: Why do we want `service` annotations to be applied here ?

File: charts/openmetadata/templates/networkpolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: openmetadata
```
Comment (akash-jain-10) [inline charts/openmetadata/templates/networkpolicy.yaml:16]: This is not required. We already have `podSelector` for the spec below.

File: charts/openmetadata/templates/networkpolicy.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  podSelector:
+    matchLabels: {{- include "OpenMetadata.selectorLabels" . | nindent 6 }}
+  policyTypes:
+  - Ingress
+    # Allow inbound connections
+  ingress:
+    - ports:
+      - port: {{ .Values.service.port }}
```
Comment (akash-jain-10) [inline charts/openmetadata/templates/networkpolicy.yaml:24]: Let's explicitly set `protocol` here to `TCP`. Check this with `kubectl explain networkpolicy.spec.ingress.ports`

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]

File: charts/openmetadata/templates/networkpolicy.yaml (hunk 1 above)
Comment (Anuj359) [reply charts/openmetadata/templates/networkpolicy.yaml:6]: Noted.

**Review by akash-jain-10 — APPROVED** [review]

**Pull request merged**